src/bootstrap/setup.rs

```diff
@@ -105,15 +129,15 @@ pub fn setup(src_path: &Path, profile: Profile) {
 // Used to get the path for `Subcommand::Setup`
 pub fn interactive_path() -> io::Result<Profile> {
     let mut input = String::new();
-    println!(
-        "Welcome to the Rust project! What do you want to do with x.py?
-a) Contribute to the standard library
-b) Contribute to the compiler
-c) Contribute to the compiler, and also modify LLVM or codegen
-d) Install Rust from source"
-    );
+    println!("Welcome to the Rust project! What do you want to do with x.py?");
+    for (letter, profile) in Profile::abbrev_all() {
+        println!("{}) {}: {}", letter, profile, profile.purpose());
+    }
     let template = loop {
-        print!("Please choose one (a/b/c/d): ");
+        print!(
+            "Please choose one ({} or keyword): ",
+            Profile::abbrev_all().map(|(l, _)| l).collect::<Vec<_>>().join("/")
+        );
         io::stdout().flush()?;
         io::stdin().read_line(&mut input)?;
```

Comment:
Pre-existing, but this is buggy - it doesn't clear the input between iterations. Do you mind fixing it here?
```
Please choose one (a/b/c/d or keyword): x
error: unknown profile: 'x'
note: press Ctrl+C to exit
Please choose one (a/b/c/d or keyword): y
error: unknown profile: 'x
y'
note: press Ctrl+C to exit
```
```suggestion
        input.clear();
        io::stdin().read_line(&mut input)?;
```

Reply:
I am fixing this by moving the definition of `input` until just before we need it.